ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -75,6 +77,9 @@ public class Transaction
 
   public static final BigInteger TWO = BigInteger.valueOf(2);
 
+  public static final Cache<Hash, Address> senderCache =
+      CacheBuilder.newBuilder().recordStats().maximumSize(100_000L).build();
```

Comment:
how much more memory will take a full cache? and the field should be private

Reply:
I made it public to expose it outside the class and the package in case we need it, but you're right, it's better to make it private and update it when need more visibility. I'll share the cache metrics in this PR.